test/OrchardCore.Tests/Shell/ShellContainerFactoryTests.cs

```diff
@@ -172,6 +172,38 @@ public async Task AssignsTypeToMultipleFeatures()
         Assert.Equal(expectedFeatureInfos, typeFeatureProvider.GetFeaturesForDependency(typeof(TestService)));
     }
 
+    [Fact]
+    public async Task ConfigureServices_NotCalled_IfRequiredFeaturesNotEnabled()
+    {
+        var shellBlueprint = CreateBlueprint();
+
+        // Only FeatureA is enabled, but the startup requires both FeatureA and FeatureB
+        var featureA = new FeatureInfo("FeatureA", "FeatureA", 1, "Tests", null, new ExtensionInfo("FeatureA"), null, false, false, false);
+        var featureB = new FeatureInfo("FeatureB", "FeatureB", 1, "Tests", null, new ExtensionInfo("FeatureB"), null, false, false, false);
+        shellBlueprint.Dependencies.Add(typeof(StartupWithRequiredFeatures), new[] { featureA });
```

Comment:
This test is not correct IMHO, `shellBlueprint` should not contain the type with missing requirements. Adding it here manually would of cause fail for the current version, it only works with the additional check you added in `ShellContainerFactory`.

Reply:
Please make the necessary changes. I am out and won't be able to make any updates anytime soon

Reply:
I'll try to add some unit tests for this case, as I'm failing to reproduce the issue. I think there must be something else going on, what @Piedone describes in https://github.com/CrestApps/CrestApps.OrchardCore/pull/136#issuecomment-3104706003 seems to be unlikely.

Reply:
The `CompositionStrategyTests` you added seems to be checking all that is required. The `ShellBlueprint.Dependencies` will contain the startup classes being run and the ones with missing requirements must not be present. This cannot be tested here on the `ShellContainerFactory`.